xmtp/src/conversations.rs

```diff
@@ -82,6 +95,146 @@ where
+                        );
+                        self.client.store.set_msg_status(
+                            transaction_manager,
+                            payload_id,
+                            status,
+                        )?;
+                    }
+                    Err(err) => {
+                        log::error!("Error processing msg: {:?}", err);
+                        return Err(StorageError::Unknown(err.to_string()));
+                    }
+                }
+            }
+
+            Ok(())
+        })?;
+
+        Ok(())
+    }
+
+    fn process_inbound_message(
+        &self,
+        conn: &mut DbConnection,
+        msg: InboundMessage,
+    ) -> Result<InboundInviteStatus, ConversationError> {
+        let message_envelope: PadlockMessageEnvelope =
+            shuck_ok!(decode_bytes(&msg.payload), InboundInviteStatus::Invalid);
+
+        let message_header: PadlockMessageHeader = shuck_ok!(
+            decode_bytes(&message_envelope.header_bytes),
+            InboundInviteStatus::Invalid
+        );
+        let unsealed_header: PadlockMessageSealedMetadata = shuck_ok!(
+            decode_bytes(&message_header.sealed_metadata),
+            InboundInviteStatus::Invalid
+        );
+
+        let plaintext = if unsealed_header.is_prekey_message {
+            let network_contact = block_on(self.client.download_contact_for_installation(
+                &unsealed_header.sender_user_address,
+                &unsealed_header.sender_installation_id,
+            ))?;
+
+            let contact = match network_contact {
+                Some(contact) => contact,
+                None => return Ok(InboundInviteStatus::Invalid),
+            };
+
+            let (_, message) =
+                self.client
+                    .create_inbound_session(conn, &contact, &message_envelope.ciphertext)?;
+
+            message
+        } else {
+            let mut session = self
+                .find_existing_session(&unsealed_header.sender_installation_id, conn)?
+                .ok_or(ConversationError::NoSessions(
+                    "Message was not PreKey message, but no existing session".into(),
+                ))?;
+
+            let olm_message: OlmMessage = shuck_ok!(
+                serde_json::from_slice(&message_envelope.ciphertext),
+                InboundInviteStatus::DecryptionFailure
+            );
+
+            session.decrypt(olm_message, conn)?
+        };
+
+        let message_obj = PadlockMessagePayload::decode(plaintext.as_slice())
+            .map_err(ConversationError::Decode)?;
+
+        let stored_message = NewStoredMessage::new(
+            message_obj.convo_id,
+            unsealed_header.sender_user_address,
+            message_obj.content_bytes,
+            MessageState::Received as i32,
+        );
+
+        self.client
+            .store
+            .insert_or_ignore_message(conn, stored_message)?;
+
+        Ok(InboundInviteStatus::Processed)
```

Comment:
Messages should probably have their own status enum, even if the values are the same as `InboundInviteStatus` today

Reply:
My intention is to merge the code paths for receiving Invites and messages, in which case the statuses would be shared across a InboundPayloadStatus enum. 

For today I'll add a copy, and we can revisit later